Source/engine/animationinfo.h

```diff
@@ -37,41 +37,46 @@ enum AnimationDistributionFlags : uint8_t {
  */
 class AnimationInfo {
 public:
-	/**
-	 * @brief Animation sprite
-	 */
-	std::optional<CelSprite> celSprite;
-	/**
-	 * @brief How many game ticks are needed to advance one Animation Frame
-	 */
-	int TicksPerFrame;
-	/**
-	 * @brief Increases by one each game tick, counting how close we are to TicksPerFrame
-	 */
-	int TickCounterOfCurrentFrame;
-	/**
-	 * @brief Number of frames in current animation
-	 */
-	int NumberOfFrames;
-	/**
-	 * @brief Current frame of animation
-	 */
-	int CurrentFrame;
-	/**
-	 * @brief Is the animation currently petrified and shouldn't advance with gfProgressToNextGameTick
-	 */
-	bool IsPetrified;
+	// clang-format off
+	// generic getters
+	[[nodiscard]] const std::optional<CelSprite>  &getCelSprite() const { return celSprite_; }
+	[[nodiscard]] int getCurrentFrame()                           const { return currentFrame_; }
+	[[nodiscard]] int getNumberOfFrames()                         const { return numberOfFrames_; }
+	[[nodiscard]] int getTickCounterOfCurrentFrame()              const { return tickCounterOfCurrentFrame_; }
+	[[nodiscard]] int getTicksPerFrame()                          const { return ticksPerFrame_; }
+	[[nodiscard]] int getTriggerFrame()                           const { return triggerFrame_; }
+
+	// helper getters
+	[[nodiscard]] bool isLastFrame()         const { return (currentFrame_ == numberOfFrames_ - 1); }
+	[[nodiscard]] bool isOver()              const { return (currentFrame_ >= numberOfFrames_ - 1); }
+	[[nodiscard]] bool isTriggerFrame()      const { return (currentFrame_ == triggerFrame_); }
+	[[nodiscard]] bool isAfterTriggerFrame() const { return (currentFrame_  > triggerFrame_); }
+
+	// generic setters
+	void setCelSprite(const std::optional<CelSprite> &celSprite) { celSprite_ = celSprite; }
+	void setCurrentFrame(const int frameNumber)                  { currentFrame_ = frameNumber; }
+	void setNumberOfFrames(const int numberOfFrames)             { numberOfFrames_ = numberOfFrames; }
+	void setTickCounterOfCurrentFrame(const int tickCounter)     { tickCounterOfCurrentFrame_ = tickCounter; }
+	void setTicksPerFrame(const int ticksPerFrame)               { ticksPerFrame_ = ticksPerFrame; } 
+	void setTriggerFrame(const int triggerFrame)                 { triggerFrame_ = triggerFrame; }

```

Comment:
This is a codestyle question, but still adding my own opinion:
I'm not in favor of adding simple getter/setter. They make the code harder to read and you have to write more code. (It would be something else if some external library would depend on our stuff.)

Small anecdote: When initially writing `AnimationInfo` I thought about making all members private and have only two methods that change/init a animation:
`SetNewAnimation` and `LoadAnimationFromSave`. But then I discovered more and more special cases. So I'm happy to see that this is picked up again.

Reply:
> So I'm happy to see that this is picked up again.

This move to private was done only to conform clang-tidy rule, that was lifted after discussion on discord.
Since it's gone, I will revert it.